### PR TITLE
Fix lazy caching

### PR DESCRIPTION
**Pull request by abidlabs**
See comment below for the underlying bug. I also used this as an opportunity to prune some dead code that is no longer needed since we've consolidated the caching code for lazy and eager modes.

Closes: https://github.com/gradio-app/gradio/issues/10118

cc @apolinario 

Confirm the fix: https://huggingface.co/spaces/abidlabs/repro-examples-single-file

### CODE REVIEW

File: gradio/helpers.py [hunk 1]
```diff
@@ -599,7 +542,7 @@ def load_from_cache(self, example_id: int) -> list[Any]:
             example_id: The id of the example to process (zero-indexed).
         """
         if self.cache_examples == "lazy":
-            if cached_index := self._get_cached_index_if_cached(example_id) is None:
+            if (cached_index := self._get_cached_index_if_cached(example_id)) is None:
```
Comment (abidlabs) [inline gradio/helpers.py:545]: This was the underlying bug. The walrus operator has a [lower order of precedence](https://peps.python.org/pep-0572/#relative-precedence-of) than every other operator except the comma, so in this case `cached_index` was being assigned a boolean value (`False`) in the `else` branch, which was being implicitly converted to a `0`, which would then lead to the first cached example always being selected.

Comment (abidlabs) [reply gradio/helpers.py:547]: dammit python

File: gradio/helpers.py [hunk 2]
```diff
@@ -448,71 +456,6 @@ async def _start_caching(self):
             else:
                 await self.cache()
 
-    def lazy_cache(self) -> None:
```
Comment (freddyaboulton) [inline gradio/helpers.py:451]: Why is this being deleted?

Comment (abidlabs) [reply gradio/helpers.py:451]: Previously, we used this method to set up the event listeners in the "lazy caching" case. However, we now use the same event listeners in the "eager" and "lazy" cases -- we just call `load_from_cache` so this function is no longer needed.

Comment (freddyaboulton) [reply gradio/helpers.py:451]: Ok I missed that previous PR

File: gradio/helpers.py (hunk 1 above)
Comment (freddyaboulton) [reply gradio/helpers.py:545]: Wow

**Review by freddyaboulton — APPROVED** [review]

**Comment by abidlabs** [comment]
Thanks for reviewing @freddyaboulton!

**Pull request merged**